### PR TITLE
feature: scroll button option for input devices

### PR DESCRIPTION
**Pull request by rkubosz**
Hello!

I missed in sway an option to set a scrolling button for my trackball Logitech M570, so I decided to implement it myself.

This patch adds support for `scroll_button` command in user's config file. Below is an example of my working config:
```
input "1133:4136:Logitech_M570" {
	scroll_method on_button_down
	scroll_button 273
}
```
In result I can scroll with holding the right button instead of default middle button.

The scroll button identifier is the same as in `libinput debug-events`.

I decided to set min and max value of button identifier to 0 and 1000, because I don't really know how the situation will be with other input devices. I am sure, that the minimum possible value must be set to 0, because libinput accepts it and disables the `on_button_down` scroll method. [(source)](https://wayland.freedesktop.org/libinput/doc/latest/group__config.html#gac95a25055b22c3631e3c10c0463ca332)

### CODE REVIEW

File: sway/commands/input/scroll_button.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		config->handler_context.input_config;
+	if (!current_input_config) {
+		return cmd_results_new(CMD_FAILURE, "scroll_button",
+			"No input device defined.");
+	}
+	struct input_config *new_config =
+		new_input_config(current_input_config->identifier);
+
+	int scroll_button = atoi(argv[0]);
+	if (scroll_button < 0 || scroll_button > 1000) {
```
Comment (emersion) [inline sway/commands/input/scroll_button.c]: Is it really necessary to check `scroll_button > 1000`?

File: sway/commands/input/scroll_button.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		free_input_config(new_config);
+		return cmd_results_new(CMD_INVALID, "scroll_button",
+				"Input out of range [1, 10]");
```
Comment (emersion) [inline sway/commands/input/scroll_button.c]: This message needs to be updated.

**Comment by emersion** [comment]
This looks pretty good to me. Can you update the man pages too?

File: sway/commands/input/scroll_button.c (hunk 1 above)
Comment (rkubosz) [reply sway/commands/input/scroll_button.c]: Probably not. In the worst case the identifier will not match any physical button and the `scroll_button` option will not work. I don't see any chance to crash sway or make it unusable by setting too big identifier.

I will modify it.

**Comment by rkubosz** [comment]
Yes, I will add appropriate paragraph to man pages.

File: sway/commands/input/scroll_button.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	struct input_config *current_input_config =
+		config->handler_context.input_config;
+	if (!current_input_config) {
+		return cmd_results_new(CMD_FAILURE, "scroll_button",
+			"No input device defined.");
+	}
+	struct input_config *new_config =
+		new_input_config(current_input_config->identifier);
+
+	int scroll_button = atoi(argv[0]);
```
Comment (SirCmpwn) [inline sway/commands/input/scroll_button.c]: Use strtol and detect errors

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**